src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -408,7 +408,7 @@ protected virtual void OutputVirtualSlots(NodeFactory factory, ref ObjectDataBui
             if (declType.HasGenericDictionarySlot())
             {
                 // Note: Canonical type instantiations always have a generic dictionary vtable slot, but it's empty
-                if (declType.IsCanonicalSubtype(CanonicalFormKind.Any))
+                if (declType.IsInterface || declType.IsCanonicalSubtype(CanonicalFormKind.Any))
```

Comment:
I was looking at what the binder did, and it was not filling the dictionary slot with anything useful.
I'll investigate this further to see if we ever fill the dictionary slot.
@davidwrighton, do you know if we ever fill the slot with a dictionary pointer? If so, under what circumstances?